Examples/Algorithms/Traccc/include/ActsExamples/Traccc/DetrayPropagator.hpp

```diff
@@ -86,42 +87,76 @@ class DetrayPropagator : public PropagatorInterface {
         {direction.x(), direction.y(), direction.z()},
         startParameters.charge());
 
-    typename propagator_t::state propagation(track, m_detrayStore->detector);
-
-    // Run the actual propagation
-    m_propagator.propagate(propagation);
-
-    // Retrieve navigation information
-    auto& inspector = propagation._navigation.inspector();
-    auto& objectTracer = inspector.template get<DetrayObjectTracer>();
-
+    // Return material
+    RecordedMaterial recordedMaterial;
     PropagationSummary summary(startParameters);
-    // Translate the objects into the steps
-    for (const auto& object : objectTracer.object_trace) {
-      // Get the position of the object
-      const auto& dposition = object.pos;
-      const auto& sfDesription = object.intersection.sf_desc;
-      const auto sf =
-          detray::tracking_surface{m_detrayStore->detector, sfDesription};
-      Acts::GeometryIdentifier geoID{sf.source()};
-      // Create a step from the object
-      Acts::detail::Step step;
-      step.position = Acts::Vector3(dposition[0], dposition[1], dposition[2]);
-      step.geoID = geoID;
-      step.navDir = object.intersection.direction ? Acts::Direction::Forward
-                                                  : Acts::Direction::Backward;
-      summary.steps.emplace_back(step);
+
+    if (!m_cfg.sterile) {
```

Comment:
Maybe this should be split into two classes at some point?

Reply:
Yes, probably that's better.

Reply:
Let's understand the matrix of propagators we need for full validation and then I will split it into logical pieaces under one `DetrayPropagators.hpp` file which can have the different flavours (we will need material interaction one as well at some point).